### ISSUE TITLE
comment_references can't resolve property access on variables

### ISSUE THREAD

**Issue by pschiffmann** [dart-lang/linter#857]
This is recognized by the linter:
```dart
/// Returns [int.isEven] of [n].
bool isEven(int n) => n.isEven;
```

But this causes a warning:
```dart
/// Returns [n.isEven].
bool isEven(int n) => n.isEven;
```

**Comment by pq** [comment]
cc @scheglov @bwilkerson

**Comment by bwilkerson** [comment]
I don't remember where, but it used to be documented that code references within square brackets could consist of either
- a single identifier where the identifier is any identifier in scope for the comment (see the spec for what is in scope in doc comments),
- two identifiers separated by a period where the first identifier is the name of a class that is in scope and the second is the name of a member declared in the class,
- a single identifier followed by a pair of parentheses where the identifier is the name of a class that is in scope (used to refer to the unnamed constructor for the class), or
- two identifiers separated by a period and followed by a pair of parentheses where the first identifier is the name of a class that is in scope and the second is the name of a named constructor (not strictly necessary, but allowed for consistency).

The format was designed to allow comments to refer to declarations using a fairly natural format. It wasn't designed to allow arbitrary expressions.

**Comment by pschiffmann** [comment]
Ok, that makes sense. Would be nice to have that list available somewhere, but the official documentation guide would probably be a better place for it. Thanks for the clarification!

**Comment by pq** [comment]
Thanks for clarifying @bwilkerson.

I took a stab at some inline docs: #861.

Thanks for the report @pschiffmann!
